packages/jest-snapshot/src/index.js

```diff
@@ -146,8 +146,12 @@ const toThrowErrorMatchingSnapshot = function(received: any, expected: void) {
       `But it didn't throw anything.`,
     );
   }
+  // eslint-disable-next-line no-new-wrappers
+  const errorMessage = new String(error.message);
+  // $FlowFixMe
+  errorMessage[Symbol.for('toThrowErrorMatchingSnapshot')] = true;
 
-  return toMatchSnapshot.call(this, error.message);
+  return toMatchSnapshot.call(this, errorMessage);
```

Comment:
We pass a string to `toMatchSnapshot` so we either need to:
1) add a symbol, so we can test for it in the snapshot serializer
2) change nothing and test for ansi characters inside the plugin (which will alter our regular snapshots too)
3) pass whole Error object and test for JestAssertionError in `toMatchSnapshot` and extract its message if necessary